### PR TITLE
Java: Add XACK stream command

### PR DESCRIPTION
**Pull request by acarbonetto**
Dependent on XREADGROUP PR: https://github.com/aws/glide-for-redis/pull/1613

Adds the [XACK](https://valkey.io/commands/xack/) command.

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 1]
```diff
@@ -493,4 +493,22 @@ CompletableFuture<Map<String, Map<String, String[][]>>> xreadgroup(
...
+     * @param key The key of the stream.
+     * @param group The consumer group name.
+     * @param ids Stream entry ID to acknowledge and purge messages.
+     * @return The number of messages that were successfully acknowledged.
+     * @example
+     *     <pre>{@code
+     * String streamId = client.xadd("mystream", Map.of("myfield", "mydata")).get();
+     * // read and process messages from streamId
+     * assert 1L == client.xack("mystream", "mygroup", new String[] {streamId}).get();
+     * // messages purged from stream
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:510]:
```suggestion
     * String entryId = client.xadd("mystream", Map.of("myfield", "mydata")).get();
     * // read and process messages from streamId
     * assert 1L == client.xack("mystream", "mygroup", new String[] {entryId}).get();
     * // message purged from stream
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 2]
```diff
@@ -3564,62 +3573,6 @@ public void xgroupCreateConsumer_xreadgroup_xgroupDelConsumer(BaseClient client)
         assertInstanceOf(RequestException.class, executionException.getCause());
     }
 
-    @SneakyThrows
-    @ParameterizedTest(autoCloseArguments = false)
-    @MethodSource("getClients")
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/SharedCommandTests.java]: was it a duplicating test?

Comment (acarbonetto) [reply java/integTest/src/test/java/glide/SharedCommandTests.java]: ignore this... it's a rebasing artefact/merge conflict.  
But yeah, somehow the test got duplicated.  The above test is the one that needs to stick around.

**Review by Yury-Fridlyand — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/StreamBaseCommands.java [hunk 3]
```diff
@@ -493,4 +493,22 @@ CompletableFuture<Map<String, Map<String, String[][]>>> xreadgroup(
...
+     *
+     * @param key The key of the stream.
+     * @param group The consumer group name.
+     * @param ids Stream entry ID to acknowledge and purge messages.
+     * @return The number of messages that were successfully acknowledged.
+     * @example
+     *     <pre>{@code
+     * String entryId = client.xadd("mystream", Map.of("myfield", "mydata")).get();
+     * // read and process messages from streamId
+     * assert 1L == client.xack("mystream", "mygroup", new String[] {entryId}).get();
```
Comment (GumpacG) [inline java/client/src/main/java/glide/api/commands/StreamBaseCommands.java:509]: I think this will fail. From my understanding, we need to run `XREADGROUP` after `XADD` in order for the entry to be added to the PEL.

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 4]
```diff
@@ -3682,6 +3691,23 @@ public void xreadgroup_return_failures(BaseClient client) {
...
+        String groupName = "group" + UUID.randomUUID();
+        String zeroStreamId = "0";
+
+        // Key exists, but it is not a stream
+        assertEquals(OK, client.set(nonStreamKey, "bar").get());
+        ExecutionException executionException =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> client.xack(nonStreamKey, groupName, new String[] {zeroStreamId}).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
```
Comment (GumpacG) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:3708]: Can you also add one test for when the streamIds is an empty array?

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 5]
```diff
@@ -3682,6 +3687,45 @@ public void xreadgroup_return_failures(BaseClient client) {
         }
     }
 
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void xack_return_failures(BaseClient client) {
```
Comment (GumpacG) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:3693]: Should rename since it doesn't return all failures

**Review by GumpacG — APPROVED** [review]

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 5 above)
Comment (acarbonetto) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:3693]: all the xack calls return failures?
